azure-servicebus-spring-boot-autoconfigure/src/main/java/com/microsoft/azure/autoconfigure/servicebus/ServiceBusAutoConfiguration.java

```diff
@@ -32,13 +34,18 @@ public ServiceBusAutoConfiguration(ServiceBusProperties properties) {
     @Scope("prototype")
     @ConditionalOnMissingBean
     public QueueClient queueClient() throws InterruptedException, ServiceBusException {
-        if (properties.getQueueConnectionString() != null && properties.getQueueReceiveMode() != null) {
-            return new QueueClient(new ConnectionStringBuilder(properties.getQueueConnectionString()),
-                    properties.getQueueReceiveMode());
+        if (properties.getConnectionString() != null && properties.getQueueName() != null &&
+                properties.getQueueReceiveMode() != null) {
+            return new QueueClient(new ConnectionStringBuilder(properties.getConnectionString(),
+                    properties.getQueueName()), properties.getQueueReceiveMode());
         }
 
-        if (properties.getQueueConnectionString() == null) {
-            LOG.error("Property azure.servicebus.queue-connection-string is not set.");
+        if (properties.getConnectionString() == null) {
+            LOG.error("Property azure.servicebus.connection-string is not set.");
```

Comment:
Looks like the property names are defined as const elsewhere. It'd be better to factor out writing this particular error condition as it happens in many places.